### PR TITLE
SONAR-6605 Replace some batch formulas by compute formulas

### PR DESCRIPTION
**Pull request by julienlancelot**
SONAR-6605 Replace some batch formulas by compute formulas

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/formula/CoreFormulaRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import static org.sonar.api.measures.CoreMetrics.STATEMENTS_KEY;
+
+public class CoreFormulaRepositoryImpl implements FormulaRepository {
+
+  private static final List<Formula> FORMULAS;
+
+  static {
+    ImmutableList.Builder<Formula> formulaList = ImmutableList.builder();
+    formulaList.add(
+      // TODO When all decorators will be moved to CE, uncomment commented lines to activate all formulas and remove formulas declaration in
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/formula/CoreFormulaRepositoryImpl.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: server/sonar-server/src/main/java/org/sonar/server/computation/formula/CoreFormulaRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      // {@link org.sonar.api.measures.CoreMetrics}
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/formula/CoreFormulaRepositoryImpl.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by SonarTech** [comment]
SonarQube analysis reported 9 issues:
* MAJOR 1 major
* MINOR 7 minor
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/4784a840e1eceff4d303fbc1f4aaf2eccf754de5/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L260) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/4784a840e1eceff4d303fbc1f4aaf2eccf754de5/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L634) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/4784a840e1eceff4d303fbc1f4aaf2eccf754de5/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L611) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/4784a840e1eceff4d303fbc1f4aaf2eccf754de5/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L199) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/4784a840e1eceff4d303fbc1f4aaf2eccf754de5/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L152) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/4784a840e1eceff4d303fbc1f4aaf2eccf754de5/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L229) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/4784a840e1eceff4d303fbc1f4aaf2eccf754de5/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L1632) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-server/src/main/java/org/sonar/server/computation/formula/CoreFormulaRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import static org.sonar.api.measures.CoreMetrics.FUNCTIONS_KEY;
+import static org.sonar.api.measures.CoreMetrics.FUNCTION_COMPLEXITY_DISTRIBUTION_KEY;
+import static org.sonar.api.measures.CoreMetrics.FUNCTION_COMPLEXITY_KEY;
+import static org.sonar.api.measures.CoreMetrics.GENERATED_LINES_KEY;
+import static org.sonar.api.measures.CoreMetrics.GENERATED_NCLOC_KEY;
+import static org.sonar.api.measures.CoreMetrics.STATEMENTS_KEY;
+
+public class CoreFormulaRepositoryImpl implements FormulaRepository {
+
+  private static final List<Formula> FORMULAS;
```
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/computation/formula/CoreFormulaRepositoryImpl.java]: You can remove the static block by simply using `ImmutableList.of()`

File: server/sonar-server/src/test/java/org/sonar/server/computation/step/ComputeFormulaMeasuresStepTest.java [hunk 4]
```diff
@@ -133,6 +133,54 @@ public void add_no_measure() throws Exception {
...
+              ).build()
+          ).build()
+      ).build();
+    treeRootHolder.setRoot(project);
+    measureRepository.addRawMeasure(1211, CoreMetrics.LINES_KEY, newMeasureBuilder().create(10));
+
+    sut.execute();
+
+    assertThat(toEntries(measureRepository.getNewRawMeasures(1))).containsOnly(entryOf(CoreMetrics.NCLOC_KEY, newMeasureBuilder().create(10)));
+    Assertions.assertThat(measureRepository.getNewRawMeasures(11)).isEmpty();
```
Comment (sns-seb) [inline server/sonar-server/src/test/java/org/sonar/server/computation/step/ComputeFormulaMeasuresStepTest.java]: this `Assertions` can be removed

**Comment by sns-seb** [comment]
some feedback and then looks good to me

File: server/sonar-server/src/main/java/org/sonar/server/computation/formula/CoreFormulaRepositoryImpl.java (hunk 3 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/computation/formula/CoreFormulaRepositoryImpl.java]: Good point !

File: server/sonar-server/src/test/java/org/sonar/server/computation/step/ComputeFormulaMeasuresStepTest.java (hunk 4 above)
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/computation/step/ComputeFormulaMeasuresStepTest.java]: How ? I need both assertThat from fest assert and guava.

**Comment by julienlancelot** [comment]
Thanks!

File: server/sonar-server/src/main/java/org/sonar/server/computation/formula/CoreFormulaRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+import static org.sonar.api.measures.CoreMetrics.FUNCTION_COMPLEXITY_DISTRIBUTION_KEY;
+import static org.sonar.api.measures.CoreMetrics.FUNCTION_COMPLEXITY_KEY;
+import static org.sonar.api.measures.CoreMetrics.GENERATED_LINES_KEY;
+import static org.sonar.api.measures.CoreMetrics.GENERATED_NCLOC_KEY;
+import static org.sonar.api.measures.CoreMetrics.STATEMENTS_KEY;
+
+public class CoreFormulaRepositoryImpl implements FormulaRepository {
+
+  private static final List<Formula> FORMULAS = ImmutableList.<Formula>of(
+    // TODO When all decorators will be moved to CE, uncomment commented lines to activate all formulas and remove formulas declaration in
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/formula/CoreFormulaRepositoryImpl.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: server/sonar-server/src/main/java/org/sonar/server/computation/formula/CoreFormulaRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    // {@link org.sonar.api.measures.CoreMetrics}
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/formula/CoreFormulaRepositoryImpl.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by SonarTech** [comment]
SonarQube analysis reported 9 issues:
* MAJOR 1 major
* MINOR 7 minor
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/7e29251de3f81b09b269f927559409760ee0fc3a/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L634) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/7e29251de3f81b09b269f927559409760ee0fc3a/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L1632) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/7e29251de3f81b09b269f927559409760ee0fc3a/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L152) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/7e29251de3f81b09b269f927559409760ee0fc3a/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L199) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/7e29251de3f81b09b269f927559409760ee0fc3a/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L229) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/7e29251de3f81b09b269f927559409760ee0fc3a/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L260) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/7e29251de3f81b09b269f927559409760ee0fc3a/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L611) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 9 issues:
* MAJOR 1 major
* MINOR 7 minor
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/077bedd2ae941d320372c7c44a170be23a1679a5/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L260) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/077bedd2ae941d320372c7c44a170be23a1679a5/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L229) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/077bedd2ae941d320372c7c44a170be23a1679a5/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L1632) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/077bedd2ae941d320372c7c44a170be23a1679a5/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L611) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/077bedd2ae941d320372c7c44a170be23a1679a5/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L199) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/077bedd2ae941d320372c7c44a170be23a1679a5/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L634) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'Metric$Builder.setFormula(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/077bedd2ae941d320372c7c44a170be23a1679a5/sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java#L152) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Pull request merged**